### DISCUSSION TITLE
What is the correct way to migrate a SUFeedURL in SwiftUI?

### DISCUSSION THREAD

**Discussion by bentumbler** [sparkle-project/Sparkle#2447] [Q&A]
I am using a static upates.xml in my info.plist and this has been working fine.
I would like to change that now, ideally to a URI which redirect such as /updatemyapp (will return a 302)

If I change this in info.plist it all works fine and updates are found but the appcast generator will not build an appcast - returns:L sparkle Error Domain=SUSparkleErrorDomain Code=1002

I see a lot about channels but thios does not seem to be their intended purpose.
I can go down the path of overriding this programmatically and in SwiftUI setFeedURL works but according to the documentation it should not be used.

It seems like I need to use the delegate but I don't really know how to to this properly for SUUpdater in SwiftUI.  Has anyone got the "right way forward" and maybe an example of how to implement this

**Comment by zorgiepoo** [comment]
> If I change this in info.plist it all works fine and updates are found but the appcast generator will not build an appcast - returns:L sparkle Error Domain=SUSparkleErrorDomain Code=1002

This should be fine but there may be an issue with generate_appcast for extension-less feeds. I fixed one issue in https://github.com/sparkle-project/Sparkle/pull/2448 -- however I am not sure it's the one you ran into. There should be a more descriptive error message accompanying what you pasted. A build from this change can be fetched from https://github.com/sparkle-project/Sparkle/actions under the attached artifacts.

You can use the delegate to provide the feed URL programmatically, but it is preferred to keep it in the Info.plist first. To provide the feed URL programmatically, when creating the `SPUStandardUpdaterController`, pass a `SPUUpdaterDelegate` conforming class (that is kept strongly referenced) to `updaterDelegate` argument. In that class, implement [-feedURLStringForUpdater:](https://sparkle-project.github.io/documentation/api-reference/Protocols/SPUUpdaterDelegate.html#/c:objc(pl)SPUUpdaterDelegate(im)feedURLStringForUpdater:)

**Reply by zorgiepoo** [reply]
With generate_appcast you can also provide the output URL via `-o` -- however it will error if your archives use multiple different feeds, so to use that option you may have to remove the older archives.
